src/main/scala/org/broadinstitute/hail/utils/StringEscapeUtils.scala

```diff
@@ -8,6 +8,37 @@ object StringEscapeUtils {
 
   def hex(ch: Char): String = Integer.toHexString(ch).toUpperCase(Locale.ENGLISH)
 
+  def escapeStringSimple(str: String, escapeChar: Char, escape: (Char) => Boolean): String = {
+    val sb = new StringBuilder
+    str.flatMap { c =>
```

Comment:
I have to run, but I'm curious how this deals with surrogate pairs encoded in UTF16 which usually yield two distinct `Char`s.